### ISSUE TITLE
getter-return marking method named 'get' as an issue

### ISSUE THREAD

**Issue by jscharett** [eslint/eslint#8919]
* **ESLint Version: 4.2.0**
* **Node Version: 4.4.3**
* **npm Version: 2.15.11**

**What parser (default, Babel-ESLint, etc.) are you using?: default**

The getter-return rule seems to catch any method that is called 'get' and does not have a return statement.  The codebase I'm working on has a number of methods that are names get that take a callback and invoke the callback asynchrously.  Its an older codebase developed before Promises were standard.  This rule seems to catch those cases, which seems incorrect as they are not true 'getters'.

Example:
var x = { get: function(callback) { setTimeout(callback, 0); }};
x.get(function () {....});

**Comment by not-an-aardvark** [comment]
I think the goal of this was to catch things like

```js
Object.defineProperty(foo, 'bar', {
    get: function() {
        // no return statement
    }
});
```

But I agree that the current check seems overly strict, since it applies to any function named "get". If it's trying to detect errors in `Object.defineProperty`, it should actually make sure the object is part of an `Object.defineProperty` call.

**Comment by ilyavolodin** [comment]
This should also check for `Object.defineProperties` as well as ES6 getter syntax.

**Comment by not-an-aardvark** [comment]
It already checks getter syntax. (See [here](https://github.com/eslint/eslint/blob/9abc6f719aed91fdbd7672f44c166a3251d44eea/lib/rules/getter-return.js#L115-L119) for how it decides whether to check a function.)

However, there is also another issue where it has a false negative for arrow functions in property descriptors:

```js
Object.defineProperty(foo, 'bar', {get: () => {}})
```

**Comment by Aladdin-ADD** [comment]
I wrote this rule and this is intended, as the saying in the PR
>for code like this:
```js
  var foo = { get: function () {bar();} };
```
>it is not a getter function, but naming a method get that does not return a value, is so confusing, causing an error may be somewhat reasonable.

In most cases, I think it may be reasonable.

**Comment by platinumazure** [comment]
I would be :+1: to only checking `get` functions that are part of `Object.defineProperty` or `Object.defineProperties` calls.
